forest/daemon/src/daemon.rs

```diff
@@ -377,8 +381,28 @@ pub(super) async fn start(config: Config, detached: bool) -> Db {
 
     services.spawn(p2p_service.run());
 
-    // Block until ctrl-c is hit
-    ctrlc_oneshot.await.unwrap();
+    // in case any services are running wait for them
+    // either to finish executing
+    // either to finish with an error and proceed to graceful shutdown
+    // or CTRL-C to be pressed and proceed to graceful shutdown
+    while !services.is_empty() {
+        select! {
+            option = services.join_next().fuse() => {
+                if let Some(Ok(Err(error_message))) = option {
+                    let msg = format!("services failure: {}", error_message);
+                    error!("{}", msg);
+                    break
+                }
+            },
+            _ = ctrlc_oneshot => break
+        }
+    }
```

Comment:
This logic is still wrong. We want to block until an error has occurred or ctrl-c has been pressed. Let's discuss if this is still unclear.